pinot-core/src/main/java/org/apache/pinot/core/query/request/context/utils/QueryContextConverterUtils.java

```diff
@@ -155,11 +157,26 @@ public static QueryContext getQueryContext(PinotQuery pinotQuery) {
       }
     }
 
+    ExplainMode explainMode;
+    if (!pinotQuery.isExplain()) {
+      explainMode = ExplainMode.NONE;
+    } else if (isUsingV1(pinotQuery)) {
+      explainMode = ExplainMode.NODE;
+    } else {
+      explainMode = ExplainMode.DESCRIPTION;
+    }
+
     return new QueryContext.Builder().setTableName(tableName).setSubquery(subquery)
         .setSelectExpressions(selectExpressions).setDistinct(distinct).setAliasList(aliasList).setFilter(filter)
         .setGroupByExpressions(groupByExpressions).setOrderByExpressions(orderByExpressions)
         .setHavingFilter(havingFilter).setLimit(pinotQuery.getLimit()).setOffset(pinotQuery.getOffset())
         .setQueryOptions(pinotQuery.getQueryOptions()).setExpressionOverrideHints(expressionContextOverrideHints)
-        .setExplain(pinotQuery.isExplain()).build();
+        .setExplain(explainMode).build();
+  }
+
+  private static boolean isUsingV1(PinotQuery pinotQuery) {
```

Comment:
Renamed and modified to delegate on QueryOptionUtils. I'm not directly using isUseMultistageEngine because I assume the map may be null, so we need to verify that first